sale_comment_template/tests/test_sale_order_report.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2017 Simone Rubino - Agile Business Group
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import odoo.tests
+from odoo.tests.common import TransactionCase
+
+
+@odoo.tests.common.at_install(False)
+@odoo.tests.common.post_install(True)
+class TestAccountInvoiceReport(TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.base_comment_model = self.env["base.comment.template"]
+        self.partner = self.env["res.partner"].create({"name": "Partner Test"})
+        # Create comment related to sale model
+        self.sale_obj = self.env.ref("sale.model_sale_order")
+        self.sale_before_comment = self._create_comment(
+            self.sale_obj, "before_lines", 1
+        )
+        self.sale_after_comment = self._create_comment(self.sale_obj, "after_lines", 2)
+        # Create comment related to move model
+        self.move_obj = self.env.ref("account.model_account_move")
+        self.move_before_comment = self._create_comment(
+            self.move_obj, "before_lines", 3
+        )
+        self.move_after_comment = self._create_comment(self.move_obj, "after_lines", 4)
+        self.sale_order = self.env.ref("sale.sale_order_7").copy(
+            {"partner_id": self.partner.id, "partner_invoice_id": self.partner.id}
+        )
+        # Trigger qty_to_invoice again
+        for order_line in self.sale_order.order_line:
+            order_line.product_id.invoice_policy = "order"
+        self.sale_order.action_confirm()
+
+    def _create_comment(self, model, position, priority):
+        return self.base_comment_model.create(
+            {
+                "name": "Comment " + position,
+                "position": position,
+                "text": "Text " + position,
+                "partner_ids": [(6, 0, self.partner.ids)],
+                "model_ids": [(6, 0, model.ids)],
+                "priority": priority,
+            }
+        )
+
+    def test_comments_in_sale_order(self):
+        res = (
+            self.env["ir.actions.report"]
+            ._get_report_from_name("sale.report_saleorder")
+            .render_qweb_html(self.sale_order.ids)
+        )
+        self.assertRegexpMatches(str(res[0]), self.sale_before_comment.text)
+        self.assertRegexpMatches(str(res[0]), self.sale_after_comment.text)
```

Comment:
The same I commented here https://github.com/OCA/account-invoice-reporting/pull/174#discussion_r608901885